packages/web/src/components/collection/desktop/CollectionHeader.tsx

```diff
@@ -125,91 +171,102 @@ export const CollectionHeader = (props: CollectionHeaderProps) => {
 
   const TitleComponent = isOwner ? 'button' : 'span'
 
+  const isPremium =
+    isStreamGated && isContentUSDCPurchaseGated(streamConditions)
+
+  const renderAlbumDetailsText = () => {
+    const releaseAndUpdatedText = lastModifiedDate
+      ? `Released ${formatDate(`${releaseDate}`)}, Updated ${formatDate(
+          `${lastModifiedDate}`
+        )}`
+      : `Released ${formatDate(`${releaseDate}`)}`
+
+    const trackCountText = `${numTracks} tracks`
+    const durationText = duration ? `, ${formatSecondsAsText(duration)}` : ''
+    return `${releaseAndUpdatedText} • ${trackCountText}${durationText}`
+  }
+
   return (
-    <Flex className={styles.collectionHeader} direction='column' gap='m'>
-      <div className={styles.topSection}>
-        <Artwork
-          collectionId={collectionId}
-          coverArtSizes={coverArtSizes}
-          callback={handleLoadArtwork}
-          gradient={gradient}
-          icon={icon}
-          imageOverride={imageOverride}
-          isOwner={isOwner}
-        />
-        <div className={styles.infoSection}>
-          <span className={cn(styles.typeLabel, fadeIn)}>
-            {!isPublished ? (
-              <IconVisibilityHidden className={styles.labelIcon} />
-            ) : null}
-            <p className={styles.label}>
-              {type === 'playlist' && !isPublished ? 'hidden playlist' : type}
-            </p>
-          </span>
-          <TitleComponent
-            className={cn(styles.title, { [styles.editableTitle]: isOwner })}
-            onClick={isOwner ? handleClickEditTitle : undefined}
-          >
-            <Text
-              variant='heading'
-              size='xl'
-              className={cn(styles.titleHeader, fadeIn)}
-            >
-              {title}
-            </Text>
-            <ClientOnly>
-              {isOwner ? <IconPencil className={styles.editIcon} /> : null}
-            </ClientOnly>
-            {isLoading ? <Skeleton className={styles.skeleton} /> : null}
-          </TitleComponent>
-          <Flex>
-            {artistName ? (
+    <Flex direction='column'>
+      {/* Top Section */}
```

Comment:
I'm in favor of more comments, but at the point the component is so long we feel the need for this, maybe we consider breaking it into smaller pieces?

Downside is the way we have it set up currently, that requires extra prop drilling and overhead, so ok with skipping it here.